api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -228,21 +229,29 @@ private void ensureDrugOrderAutoExpirationDateIsSet(Order order) {
 			((DrugOrder) order).setAutoExpireDateBasedOnDuration();
 		}
 	}
-
+	
 	private void ensureOrderTypeIsSet(Order order, OrderContext orderContext) {
 		if (order.getOrderType() != null) {
 			return;
 		}
 		OrderType orderType = null;
+		List<OrderType> orderTypes = new ArrayList<>();
 		if (orderContext != null) {
 			orderType = orderContext.getOrderType();
+			orderTypes.add(orderType);
 		}
 		if (orderType == null) {
 			orderType = getOrderTypeByConcept(order.getConcept());
 		}
+		if((orderTypes.size() > 1) && order instanceof DrugOrder){
+				throw new AmbiguousOrderException("Order.cannot.have.more.than.one");
+		}
```

Comment:
I understand you were seeing errors with these lines; however, these should only cause a problem if you test dataset has duplicate order types (i.e., a single order type returns >1 result).